lambdas/check_new_areas/function.py

```diff
@@ -1,20 +1,138 @@
 
-def handler(event, context):
-    env = event["env"]
+    now: datetime = datetime.now()
 
     try:
-        s3_client = boto3.client("s3")
-        if s3_directory_exists(NEW_AOI_BUCKET, NEW_AOI_PREFIX, s3_client):
-            aoi_url = "s3://{}/{}/*.tsv".format(NEW_AOI_BUCKET, NEW_AOI_PREFIX)
-            return {
-                "status": "FOUND_NEW"
-            }.update(event)
-    except ClientError:
-        slack_webhook("INFO", "No new user areas found. Doing nothing.")
-        return {"status": "NOTHING_TO_DO"}
\ No newline at end of file
+        areas: Dict[str, Any] = get_pending_areas()
+        geostore_ids: List[str] = get_geostore_ids(areas)
+        geostore: Dict[str, Any] = get_geostore(geostore_ids)
+
+        with geostore_to_wkb(geostore) as wkb:
+            S3_CLIENT.upload_fileobj(
+                wkb,
+                f"gfw-pipelines{bucket_suffix()}",
+                f"geotrellis/features/geostore/aoi-{now.strftime('%Y%m%d')}.tsv",
+            )
+
+        return {"status": "FOUND_NEW"}.update(event)
+    except EmptyResponseException:
+        # slack_webhook("INFO", "No new user areas found. Doing nothing.")
+        return {"status": "NOTHING_TO_DO"}
+    except Exception:
+        return {"status": "FAILED"}
+
+
+def get_pending_areas() -> Dict[str, Any]:
+    headers: Dict[str, str] = {"Authorization": f"Bearer {TOKEN}"}
+    url: str = f"http://{api_prefix()}-api.globalforestwatch.org/v2/area?status=pending&all=true"
+    r: Response = requests.get(url, headers=headers)
+
+    if not r.json():
+        raise EmptyResponseException("No pending areas")
+    else:
+        return r.json()
+
+
+def get_geostore_ids(areas: Dict[str, Any]) -> List[str]:
+
+    geostore_ids: List[str] = list()
+    for area in areas["data"]:
+        geostore_ids.append(area["attributes"]["geostore"])
+    return geostore_ids
+
+
+def get_geostore(geostore_ids: List[str]) -> Dict[str, Any]:
+    headers: Dict[str, str] = {"Authorization": f"Bearer {TOKEN}"}
+    url: str = f"https://{api_prefix()}-api.globalforestwatch.org/v1/geostore/find-by-ids"
+    r: Response = requests.post(url, data=json.dumps(geostore_ids), headers=headers)
+    return r.json()
+
+
+@contextmanager
+def geostore_to_wkb(geostore: Dict[str, Any]) -> Iterator[io.StringIO]:
+    glad_tiles: List[Polygon] = _get_glad_extent()
+    wkb = io.StringIO()
+    for g in geostore["data"]:
+        geom: Polygon = shape(
+            g["geostore"]["data"]["attributes"]["geojson"]["features"][0]["geometry"]
+        )
+        for tile in glad_tiles:
+            intersection = geom.intersection(tile)
+            if intersection.area:
+
+                wkb.write(f"{g['geostoreId']}\t{dumps(intersection, hex=True)}\n")
+    yield wkb
+
+
+def _get_glad_extent() -> List[Polygon]:
+    response: Dict[str, Any] = S3_CLIENT.get_object(
+        Bucket=f"gfw-data-lake{bucket_suffix()}",
+        Key="umd_glad_alerts/latest/vector/umd_glad_alerts__extent_1x1.geojson",
```

Comment:
Will need to review final location of this file. It represent the extent of Glad alerts sliced into 1x1 degree tiles